packages/main/src/plugin/kubernetes-client.ts

```diff
@@ -1311,4 +1319,304 @@ export class KubernetesClient {
+      currentReplicas = currentStatefulSet.spec?.replicas || 1;
+    }
+
+    await this.scaleController(appsApi, namespace, controllerName, controllerType, 0);
+
+    await new Promise(resolve => setTimeout(resolve, timeout));
+
+    await this.scaleController(appsApi, namespace, controllerName, controllerType, currentReplicas);
+  }
+
+  protected async scaleController(
+    appsApi: AppsV1Api,
+    namespace: string,
+    controllerName: string,
+    controllerType: 'Deployment' | 'ReplicaSet' | 'StatefulSet',
+    replicas: number,
+  ): Promise<void> {
+    if (controllerType === 'Deployment') {
+      await appsApi.patchNamespacedDeploymentScale(
+        controllerName,
+        namespace,
+        { spec: { replicas } },
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        { headers: { 'Content-Type': 'application/strategic-merge-patch+json' } },
+      );
+    } else if (controllerType === 'ReplicaSet') {
+      await appsApi.patchNamespacedReplicaSetScale(
+        controllerName,
+        namespace,
+        { spec: { replicas } },
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        { headers: { 'Content-Type': 'application/strategic-merge-patch+json' } },
+      );
+    } else if (controllerType === 'StatefulSet') {
+      await appsApi.patchNamespacedStatefulSetScale(
+        controllerName,
+        namespace,
+        { spec: { replicas } },
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        { headers: { 'Content-Type': 'application/strategic-merge-patch+json' } },
+      );
+    }
+  }
+
+  protected async restartJob(name: string, namespace: string): Promise<void> {
+    const batchApi = this.kubeConfig.makeApiClient(BatchV1Api);
+    const coreApi = this.kubeConfig.makeApiClient(CoreV1Api);
+
+    const { body: existingJob } = await batchApi.readNamespacedJob(name, namespace);
+    await batchApi.deleteNamespacedJob(name, namespace, 'true', undefined, undefined, undefined, 'Background');
+
+    const isJobDeleted = await this.waitForJobDeletion(batchApi, name, namespace);
+    if (!isJobDeleted) {
+      throw new Error(`job "${name}" in namespace "${namespace}" was not deleted within the expected timeframe`);
+    }
+
+    const labelSelector = `job-name=${name}`;
+    const isPodsDeleted = await this.waitForPodsDeletion(coreApi, namespace, labelSelector);
+    if (!isPodsDeleted) {
+      throw new Error(
+        `not all pods with selector "${labelSelector}" in namespace "${namespace}" were deleted within the expected timeframe`,
+      );
+    }
+
+    delete existingJob.metadata!.creationTimestamp;
+    delete existingJob.metadata!.resourceVersion;
+    delete existingJob.metadata!.selfLink;
+    delete existingJob.metadata!.uid;
+    delete existingJob.metadata!.ownerReferences;
+    delete existingJob.status;
+    delete existingJob.spec!.selector;
```

Comment:
```suggestion
  _.omit(existingJob.metadata!, ['creationTimestamp', 'resourceVersion', 'selfLink', 'uid', 'ownerReferences', 'status', 'selector']);
```

you have to install @types/lodash package as `yarn -D add @types/lodash -W` for code above to work.
the same applies to the code below.

Reply:
not really sure, that we need to install this library just for one stylish usecase